controllers/compaction/reconciler.go

```diff
@@ -96,23 +96,84 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}, err
 	}
 
-	if !etcd.DeletionTimestamp.IsZero() {
+	if !etcd.DeletionTimestamp.IsZero() || etcd.Spec.Backup.Store == nil {
 		// Delete compaction job if exists
 		return r.delete(ctx, r.logger, etcd)
 	}
 
-	if etcd.Spec.Backup.Store == nil {
-		return ctrl.Result{}, nil
+	logger := r.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String())
+
+	return r.reconcileJob(ctx, logger, etcd)
+}
+
+func (r *Reconciler) reconcileJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
+	// Update metrics for currently running compaction job, if any
+	job := &batchv1.Job{}
+	if err := r.Get(ctx, types.NamespacedName{Name: etcd.GetCompactionJobName(), Namespace: etcd.Namespace}, job); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("Currently, no compaction job is running")
```

Comment:
Can you please improve this log a bit by also logging the namespace and job name? Since druid runs for multiple etcd clusters, this log will not be very useful without additional information.

Reply:
Done. Though we shouldn't print name of the job when there is no job running.

Reply:
I think namespace and etcd name should be sufficient, and rather necessary to identify which job does not exist according to the log. It's easier to simply add a `logger.WithValues()` at the beginning of reconciliation, so that all logs already have the etcd name and namespace, and the same logger is passed around within the reconciler.